chain/x/zoracle/internal/types/code.go

```diff
@@ -0,0 +1,25 @@
+package types
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/tendermint/tendermint/crypto"
+)
+
+// StoredCode store actual code with owner
+type StoredCode struct {
+	Code  []byte
+	Owner sdk.AccAddress
+}
+
+// NewStoredCode is a constructor of StoredCode
+func NewStoredCode(code []byte, owner sdk.AccAddress) StoredCode {
+	return StoredCode{
+		Code:  code,
+		Owner: owner,
+	}
+}
+
+// GetCodeHash is a function to calculate hash of actual code
+func (c StoredCode) GetCodeHash() []byte {
```

Comment:
Just to confirm. Even though this is pass by value, it's efficient because `c.Code` is a slice and Go does not do copy on slice, correct?

Reply:
yep!